File: src/main/go/leetcode/lists/hot100/LeetCode621.go
package hot100

func leastInterval(tasks []byte, n int) int {

	if len(tasks) == 0 {
		return 0
	}
	dict := [26]int{}
	for _, ch := range tasks {
		dict[ch-'A']++
	}

	maxCount, maxCountSame := 0, 0
	for _, num := range dict {
		if num > maxCount {
			maxCount = num
			maxCountSame = 1
		} else if num == maxCount {
			maxCountSame++
		}
	}
	return Max((maxCount-1)*(n+1)+maxCountSame, len(tasks))
}

File: java/src/leetcode/LeetCode621.java
package leetcode;

/**
 * @author jingxinwu
 * @date 2021-12-18 10:21 PM
 */
public class LeetCode621 {

    //https://www.jiakaobo.com/leetcode/621.%20Task%20Scheduler.html
    public int leastInterval(char[] tasks, int n) {
        int[] dict = new int[26];
        for (char task : tasks) {
            dict[task - 'A']++;
        }
        int maxCount = 0;
        int maxCountSame = 0;
        for (int num : dict) {
            if (num > maxCount) {
                maxCount = num;
                maxCountSame = 1;
            } else if (num == maxCount) {
                maxCountSame++;
            }
        }
        int res = (maxCount - 1) * (n + 1) + maxCountSame;
        return Math.max(res, tasks.length);
    }

}
